lms/djangoapps/courseware/model_data.py

```diff
@@ -960,7 +960,7 @@ def fetch_scores(self, locations):
         # attached to them (since old mongo identifiers don't include runs).
         # So we have to add that info back in before we put it into our lookup.
         self._locations_to_scores.update({
-            UsageKey.from_string(location).map_into_course(self.course_key): self.Score(correct, total, created)
+            UsageKey.from_string(unicode(location)).map_into_course(self.course_key): self.Score(correct, total, created)
```

Comment:
I think this change may be doing more work than it needs to. I'd expect `location` here to actually start off as a UsageKey in the new paradigm.

Reply:
👍  If so, it'd be great to remove the unneeded code.